streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -296,7 +296,8 @@ public StreamThread(TopologyBuilder builder,
         this.processId = processId;
         this.partitionGrouper = config.getConfiguredInstance(StreamsConfig.PARTITION_GROUPER_CLASS_CONFIG, PartitionGrouper.class);
         this.streamsMetadataState = streamsMetadataState;
-        threadClientId = clientId + "-" + threadName;
+        //clientId may already contain application id, removing to avoid duplication
+        threadClientId = clientId.replace(applicationId, "") + "-" + threadName;
```

Comment:
The duplicate `appId` as mentioned in the JIRA comes from `threadName` (not `clientId`). As we add `appId` in the call to `super(...)`, it is contained in return value from `getName()` (line 291). Thus, it might make more sense to remove the duplicate in line 291 (we should also remove the added `"-"`, see below).

Please also make the variable `final` in line 291:

`final String threadName = getName().replace(applicationId + "-", "");`